### PR TITLE
speed and weight improvements

### PR DESCRIPTION
**Pull request by TriplEight**
- cleaned unused dependencies

Tests and build run successfully.
Please tell if I've deleted something important.

### CODE REVIEW

File: docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile [hunk 1]
```diff
@@ -11,25 +14,35 @@ LABEL maintainer="devops-team@parity.io" \
...
 	apt-get -y update; \
 	apt-get install -y --no-install-recommends \
-		software-properties-common curl git \
-		make cmake g++ clang rhash \
-		pkg-config libudev-dev time \
-		libssl-dev; \
+		wget git ca-certificates gcc g++ clang \
+		pkg-config libssl-dev cmake libc6-dev; \
+# needed: wget ca-certificates gcc g++ libssl-dev pkg-config git cmake clang
+# not needed: software-properties-common curl make rhash libudev-dev time
```
Comment (TriplEight) [inline docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile]: @tomaka @bkchr is it okay to remove `make rhash libudev-dev`? They didn't show up during build and test.

**Comment by gabreal** [comment]
how do the **ci/dockercloud** status updates arrive here?

File: docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile [hunk 2]
```diff
@@ -15,10 +16,8 @@ WORKDIR /builds/parity/substrate/
 RUN set -eux; \
 	apt-get -y update; \
 	apt-get install -y --no-install-recommends \
-		software-properties-common curl git \
-		make cmake g++ clang rhash \
-		pkg-config libudev-dev time \
-		libssl-dev; \
+		git g++ clang cmake make \
+		pkg-config libssl-dev; \
```
Comment (gabreal) [inline docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile]: I completely agree with optimization but I consider *curl* like an essential package that can be useful in so many situations that I would leave it there (but I'm also fine having it removed).
**time** will certainly be needed for shell scripts that don't require bash.
@tomaka what about clang isn't it still needed in the builds of polkadot or substrate?

Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile]: `time` is built in, as well as `ca-certificates` and `gcc`.
OK, I'll include `curl`

**Comment by TriplEight** [comment]
> how do the **ci/dockercloud** status updates arrive here?

https://cloud.docker.com/repository/docker/parity/rust-centos/builds/edfc804e-e2b7-46b6-81b7-2fa214feffcd

i have no idea yet

File: docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile (hunk 2 above)
Comment (gabreal) [reply docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile]: please verify on systems that don't have `/bin/sh` linked to bash e.g. Debian.

Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile]: actually:
```
The following NEW packages will be installed:
  clang clang-3.8 cmake cmake-data curl g++ g++-6 git git-man libarchive13
  libbsd0 libclang-common-3.8-dev libclang1-3.8 libcurl3 libcurl3-gnutls
  libdpkg-perl libedit2 liberror-perl libexpat1 libffi6 libgc1c2 libgdbm3
  libglib2.0-0 libgnutls30 libgssapi-krb5-2 libhogweed4 libicu57 libidn11
  libidn2-0 libjsoncpp1 libk5crypto3 libkeyutils1 libkrb5-3 libkrb5support0
  libldap-2.4-2 libldap-common libllvm3.8 liblzo2-2 libncurses5 libnettle6
  libnghttp2-14 libobjc-6-dev libobjc4 libp11-kit0 libperl5.24 libprocps6
  libpsl5 librtmp1 libsasl2-2 libsasl2-modules-db libssh2-1 libssl-dev
  libssl1.0.2 libstdc++-6-dev libtasn1-6 libunistring0 libuv1 libxml2 make
  perl perl-modules-5.24 pkg-config procps
```
`clang` is installed

Comment (gabreal) [reply docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile]: Okay sorry I was blind :see_no_evil: about the clang thing - sorry @tomaka nothing to see here :)

Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile]: should I move the changes to the branch with more appropriate name?

File: docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile (hunk 1 above)
Comment (tomaka) [reply docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile]: I don't know what `rhash` is, but yeah we're not using `libudev-dev` in Substrate.

File: docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile (hunk 2 above)
Comment (gabreal) [reply docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile]: I think the branch name won't be used anymore after merging. so for me it's fine.

File: docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile (hunk 1 above)
Comment (kirushik) [reply docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile]: AFAIR, we use `rhash` to calculate hashes of the binaries we've built. It's just a faster version of  `sha256sum`, and is called from our release scripts. I think it should stay.

`libudev-dev` is used when we build, say, Parity Ethereum with hardware wallets support — but I'm not sure if Substrate supports hardware wallets now, or is it planned short-term

Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile]: This is why I asked.
We use `rhash` to sign releases in Parity Ethereum, and `sha256sum` in Substrate.
Should I change the sign to `rhash`?

Comment (gabreal) [reply docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile]: IMO `sha256sum` is most common and most people already have it installed, so if there is no important reason I suggest to stick to that. A speed-up is not mentioned on their [github page](https://github.com/rhash/RHash).

Comment (gabreal) [reply docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile]: Okay it's faster :)
```
% cat grml64-full_2018.12.iso > /dev/null
% time sha256sum grml64-full_2018.12.iso
33617aba115ccb6f99bd59406ebe04ffdc0b9ac15a997c0bd3800b8ad6c7dc91  grml64-full_2018.12.iso
sha256sum grml64-full_2018.12.iso  2.90s user 0.11s system 99% cpu 3.006 total
% time rhash --sha256 grml64-full_2018.12.iso
33617aba115ccb6f99bd59406ebe04ffdc0b9ac15a997c0bd3800b8ad6c7dc91  grml64-full_2018.12.iso
rhash --sha256 grml64-full_2018.12.iso  1.49s user 0.12s system 99% cpu 1.610 total
% time sha256sum grml64-full_2018.12.iso
33617aba115ccb6f99bd59406ebe04ffdc0b9ac15a997c0bd3800b8ad6c7dc91  grml64-full_2018.12.iso
sha256sum grml64-full_2018.12.iso  2.84s user 0.10s system 99% cpu 2.951 total
```

Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/substrate-linux-ci/Dockerfile]: it makes sense then to start using it in Substrate as well.

**Pull request merged**